src/overlays/actors/ovl_Obj_Grass/z_obj_grass.c

```diff
@@ -28,41 +35,495 @@ const ActorInit Obj_Grass_InitVars = {
     (ActorFunc)ObjGrass_Draw,
 };
 
-// static ColliderCylinderInit sCylinderInit = {
-static ColliderCylinderInit D_809AAB20 = {
-    { COLTYPE_NONE, AT_NONE, AC_ON | AC_TYPE_PLAYER, OC1_ON | OC1_TYPE_PLAYER | OC1_TYPE_2, OC2_TYPE_2, COLSHAPE_CYLINDER, },
-    { ELEMTYPE_UNK0, { 0x00000000, 0x00, 0x00 }, { 0x0580C71C, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_ON, OCELEM_ON, },
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_ON | AC_TYPE_PLAYER,
+        OC1_ON | OC1_TYPE_PLAYER | OC1_TYPE_2,
+        OC2_TYPE_2,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK0,
+        { 0x00000000, 0x00, 0x00 },
+        { 0x0580C71C, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_ON,
+        OCELEM_ON,
+    },
     { 6, 44, 0, { 0, 0, 0 } },
 };
 
-#endif
+Vec3f D_809AAB4C[] = {
+    { 0.0f, 0.707099974155f, 0.707099974155f },
+    { 0.707099974155f, 0.707099974155f, 0.0f },
+    { 0.0f, 0.707099974155f, -0.707099974155f },
+    { -0.707099974155f, 0.707099974155f, 0.0f },
+};
+
+s16 D_809AAB7C[] = { 0x6C, 0x66, 0x60, 0x54, 0x42, 0x37, 0x2A, 0x26 };
+
+s32 func_809A9110(GlobalContext* globalCtx, Vec3f* arg1) {
+    f32 sp2C;
+    Vec3f sp20;
+
+    SkinMatrix_Vec3fMtxFMultXYZW(&globalCtx->viewProjectionMtxF, arg1, &sp20, &sp2C);
+
+    if ((globalCtx->unk_187F0.z * -130.13191f) < sp20.z) {
+        if (sp2C < 1.0f) {
+            sp2C = 1.0f;
+        }
+
+        if (((fabsf(sp20.x) - (130.13191f * globalCtx->unk_187F0.x)) < sp2C) &&
+            ((fabsf(sp20.y) - (130.13191f * globalCtx->unk_187F0.y)) < sp2C)) {
+            return true;
+        }
+    }
+    return false;
+}
+
+void func_809A91FC(MtxF* matrix) {
+    s32 i;
+    MtxF* temp = Matrix_GetCurrentState();
+    f32* tmp = &temp->mf[0][0];
+    f32* tmp2 = &matrix->mf[0][0];
 
-extern ColliderCylinderInit D_809AAB20;
+    for (i = 0; i < 16; i++) {
+        *tmp++ += *tmp2++;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Grass/func_809A9110.s")
+void func_809A92D0(ObjGrassStruct1_1* ptr, GlobalContext* globalCtx) {
+    if (!(ptr->unk_0E & 0x10)) {
+        Item_DropCollectibleRandom(globalCtx, NULL, &ptr->unk_00, ptr->unk_0E * 0x10);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Grass/func_809A91FC.s")
+#ifdef NON_MATCHING
```

Comment:
```suggestion
#ifdef NON_EQUIVALENT
```
NON_MATCHING and NON_EQUIVALENTS cannot be mixed due to relocs. This should be changed to a NON_EQUIVALENT, though should also leave a comment that is just NON_MATCHING

Reply:
shouldn't it just stay NON_MATCHING if it's NON_MATCHING?

Reply:
The problem is leaving this as NON_MATCHING would break NON_MATCHING builds since the correct relocs can't be used when there is a NON_EQUIVALENT as well.

Reply:
Makes sense, but then won't NON_EQUIVALENT builds be broken for the same reason?

Reply:
Technically yes, but we don't test or use NON_EQUIVALENT builds.